python_modules/libraries/dagster-airlift/dagster_airlift/core/monitoring_job/event_stream.py

```diff
@@ -254,6 +260,37 @@ def _process_completed_runs(
             break
 
 
+async def _retrieve_logs_for_task_instance(
+    context: OpExecutionContext,
+    airflow_instance: AirflowInstance,
+    task_instance: TaskInstance,
+) -> TaskInstanceCompleted:
+    logs = airflow_instance.get_task_instance_logs(
+        task_instance.dag_id,
+        task_instance.task_id,
+        task_instance.run_id,
+        task_instance.try_number,
+    )
+    metadata = extract_metadata_from_logs(context, logs)
```

Comment:
this might be the right place to toss a try/except, haven't thought through it all the way

Reply:
I think we probably want to wrap in two places: one for the json load, and one for the metadata coercion